File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        while l < r:
            mid = (l+r)//2
            if nums[mid] < nums[len(nums)-1]:
                r = mid
            else:
                l = mid + 1
        return nums[l]

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.c
int findMin(int* nums, int numsSize) {
    int l = 0;
    int r = numsSize-1;
    int middle;
    while (l<r){
        if (nums[l]<nums[r]){
            return nums[l];
        }

        middle = (l+r)/2;
        if (nums[middle]<nums[r]){
            r = middle;
        }
        else{
            l = middle+1;
        }
    }
    return nums[l];
}